src/test/regress/expected/subselect_gp.out

```diff
@@ -3419,27 +3429,324 @@ with cte as
 (delete from t1
  returning *)
 select * from t_repl
-where t_repl.i in (select i from cte where t_repl.i = cte.i);
+where t_repl.i in (select i from cte where t_repl.i = cte.i and cte.i > 0);
                              QUERY PLAN                             
 --------------------------------------------------------------------
  Gather Motion 1:1  (slice2; segments: 1)
    ->  Seq Scan on t_repl
          Filter: (SubPlan 1)
-         SubPlan 1  (slice2; segments: 3)
-           ->  Subquery Scan on cte
+         SubPlan 1  (slice2; segments: 1)
+           ->  Result
                  Filter: (t_repl.i = cte.i)
                  ->  Materialize
                        ->  Gather Motion 3:1  (slice1; segments: 3)
-                             ->  Delete on t1
-                                   ->  Seq Scan on t1
+                             ->  Subquery Scan on cte
+                                   Filter: (cte.i > 0)
+                                   ->  Delete on t1
+                                         ->  Seq Scan on t1
  Optimizer: Postgres query optimizer
-(11 rows)
+(13 rows)
+
+with cte as
+(delete from t1
+ returning *)
+select * from t_repl
+where t_repl.i in (select i from cte where t_repl.i = cte.i and cte.i > 0);
+ i 
+---
+ 1
+(1 row)
+
+select count(*) from t1;
+ count 
+-------
+     0
+(1 row)
+
+--start_ignore
+drop table if exists t3;
+NOTICE:  table "t3" does not exist, skipping
+--end_ignore
+create table t3 (i int, j int) distributed randomly;
```

Comment:
Can you explain the meaning of this additional tests?

Reply:
So it's a way to get a trivial SubqueryScan without quals that will be thrown away during setting references.